pkg/util/test/util.go

```diff
@@ -2951,6 +2955,177 @@ func ValidatePrometheus(pxImageList map[string]string, cluster *corev1.StorageCl
+		logrus.Infof("Skipping grafana validation for operation version: [%s]", opVersion.String())
+		return nil
+	}
+
+	shouldBeInstalled := cluster.Spec.Monitoring != nil && cluster.Spec.Monitoring.Grafana != nil &&
+		cluster.Spec.Monitoring.Grafana.Enabled && cluster.Spec.Monitoring.Prometheus != nil &&
+		cluster.Spec.Monitoring.Prometheus.Enabled
+	err = ValidateGrafanaDeployment(cluster, shouldBeInstalled, pxImageList)
+	if err != nil {
+		return err
+	}
+	err = ValidateGrafanaService(shouldBeInstalled)
+	if err != nil {
+		return err
+	}
+	err = ValidateGrafanaConfigmaps(shouldBeInstalled)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func ValidateGrafanaDeployment(cluster *corev1.StorageCluster, shouldBeInstalled bool, pxImageList map[string]string) error {
+	check := func() (interface{}, bool, error) {
+		pods, err := coreops.Instance().GetPods("kube-system", map[string]string{"app": "grafana"})
+		if err != nil {
+			return nil, true, err
+		}
+		if shouldBeInstalled {
+			if len(pods.Items) == 0 || pods.Items[0].Status.Phase != v1.PodRunning {
+				logrus.Errorf("Grafana not yet ready")
+				return "", true, fmt.Errorf("grafana is not installed when it should be")
+			}
+
+			err := validateContainerImageInsidePods(cluster, pxImageList["grafana"], "grafana", pods)
+			if err != nil {
+				logrus.Errorf("Grafana image is invalid yet ready")
+				return "", true, fmt.Errorf("grafana is not installed when it should be")
+			}
+
+			logrus.Infof("Grafana installed successfully")
+			return "", false, nil
+		} else {
+			if len(pods.Items) < 1 {
+				logrus.Infof("Grafana uninstalled successfully")
+				return "", false, nil
+			} else {
+				logrus.Errorf("Grafana not yet uninstalled")
+				return "", true, fmt.Errorf("grafana is installed when it is expected to be uninstalled")
+			}
+		}
+	}
+
+	if _, err := task.DoRetryWithTimeout(check, 2*time.Minute, 30*time.Second); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func ValidateGrafanaDeploymentImage(image string) error {
+	check := func() (interface{}, bool, error) {
+		pods, err := coreops.Instance().GetPods("kube-system", map[string]string{"app": "grafana"})
+		if err != nil {
+			return "", true, err
+		}
+
+		podCount := len(pods.Items)
+		if podCount > 0 && len(pods.Items[0].Spec.Containers) > 0 {
+			actualImage := pods.Items[0].Spec.Containers[0].Image
+			if actualImage == image {
+				logrus.Infof("Grafana installed successfully with image %s", image)
+				return "", false, nil
+			} else {
+				logrus.Errorf("Grafana not yet ready with image %s", image)
+				return "", true, fmt.Errorf("grafana is not installed with expected image %s. actual: %s", image, actualImage)
+			}
+		} else {
+			return "", true, fmt.Errorf("grafana is not installed with image %s. number of pods: %v", image, podCount)
+		}
+	}
+
+	if _, err := task.DoRetryWithTimeout(check, 1*time.Minute, 10*time.Second); err != nil {
+		return err
+	}
+
+	return nil
+}
```

Comment:
Remove this since you using `validateContainerImageInsidePods()` and we don't need this any more